src/components/ProgressIndicator/index.js

```diff
@@ -30,19 +32,30 @@ const ProgressIndicator = ({
   const applicableCompleted =
     hasSubmission && applicableSteps.length === applicableAndCompletedSteps.length;
 
+  // meaningful titles based on steps
+  const config = useContext(ConfigContext);
+  let currentStepTitle = '';
+
   // figure out the title for the mobile menu based on the state
   let activeStepTitle;
+
   if (isStartPage) {
     activeStepTitle = STEP_LABELS.login;
+    currentStepTitle = STEP_LABELS.login.props.defaultMessage[0].value;
   } else if (summaryMatch) {
     activeStepTitle = STEP_LABELS.overview;
+    currentStepTitle = STEP_LABELS.overview.props.defaultMessage[0].value;
   } else if (confirmationMatch) {
     activeStepTitle = STEP_LABELS.confirmation;
+    currentStepTitle = STEP_LABELS.confirmation.props.defaultMessage[0].value;
   } else {
     const step = steps.find(step => step.slug === stepSlug);
     activeStepTitle = step.formDefinition;
   }
```

Comment:
This doesn't seem right for a couple of reasons:

1. `STEP_LABELS.foo` are `FormattedMesage` components and the internal API is private. Better would be to instead change that definition to use `intl.defineMessages` so we can get the string variant in the calling code by using `intl.formatMessage`
2. The progress-indicator is not guaranteed to be rendered, it can be disabled at the form configuration level, so this is the wrong place to hook into setting the title.

Instead, I'd expect the following components to be updated to set the title:

* `FormStart`
* `Summary`
* `SubmissionConfirmation`
* `PaymentOverview`

These are the components inside of each `Route` in the `Form` component